processor/servicegraphprocessor/processor.go

```diff
@@ -254,10 +256,10 @@ func (p *processor) aggregateMetrics(ctx context.Context, td ptrace.Traces) (err
 	return nil
 }
 
-func (p *processor) upsertDimensions(m map[string]string, resourceAttr pcommon.Map, spanAttr pcommon.Map) {
+func (p *processor) upsertDimensions(kind string, m map[string]string, resourceAttr pcommon.Map, spanAttr pcommon.Map) {
 	for _, dim := range p.config.Dimensions {
 		if v, ok := findAttributeValue(dim, resourceAttr, spanAttr); ok {
-			m[dim] = v
+			m[kind+"_"+dim] = v
```

Comment:
Isn't this effectively changing the key name for the label? People relying on the current label would get affected by this change, won't they?

Reply:
> Isn't this effectively changing the key name for the label? People relying on the current label would get affected by this change, won't they?

+1

Reply:
you are right, it is a breaking change.

Reply:
so, you.d better change the `change_type ` to `breaking ` in changelog.

Reply:
Before moving forward with this change, I would like to brainstorm a bit on how we can make this with the lowest impact possible on current users of this processor.

We should add new metrics with the prefixes instead of changing the existing ones. In the changelog, mark the old ones as deprecated, and in the next release, we will remove the old metric. This way, users can read the changelog and apply changes without having their dashboards broken.

What do you think, @mapno?

Reply:
Having both labels for a while should easy the transition from one set of labels to the other. I'm in favor of supporting that for a while 👍